crypto/mem.c

```diff
@@ -165,13 +166,30 @@ void ossl_malloc_setup_failures(void)
 {
     const char *cp = getenv("OPENSSL_MALLOC_FAILURES");
 
-    if (cp != NULL && (md_failstring = strdup(cp)) != NULL)
+    if (cp != NULL) {
+        md_fs_orig = OPENSSL_strdup(cp);
```

Comment:
Please revert this to strdup(). The strdup() function was used here before already. Also calling any OPENSSL_ allocation function here makes it impossible to change the memory callbacks.

Reply:
Sorry, that was my fault - I asked for that change